### PR TITLE
Migrate more tool unit tests to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/context.dart [hunk 1]
```diff
@@ -118,7 +118,7 @@ class AppContext {
     if (value == null && _parent != null) {
       value = _parent!.get<T>();
     }
-    return _unboxNull(value ?? _generateIfNecessary(T, _fallbacks)) as T;
+    return _unboxNull(value ?? _generateIfNecessary(T, _fallbacks)) as T?;
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/base/context.dart:121]: Also in https://github.com/flutter/flutter/pull/79985, let's see which one merges first.

File: packages/flutter_tools/lib/src/dart/language_version.dart [hunk 2]
```diff
@@ -48,7 +48,7 @@ bool _inUnitTest() {
 ///
 /// The specification for the language version tag is defined at:
 /// https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/feature-specification.md#individual-library-language-version-override
-LanguageVersion determineLanguageVersion(File file, Package package, String flutterRoot) {
+LanguageVersion determineLanguageVersion(File file, Package? package, String flutterRoot) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/dart/language_version.dart:51]: This is the only actual non-test change.

File: packages/flutter_tools/test/general.shard/linux/linux_doctor_test.dart [hunk 3]
```diff
@@ -9,7 +9,7 @@ import 'package:flutter_tools/src/doctor_validator.dart';
 import 'package:flutter_tools/src/linux/linux_doctor.dart';
 
 import '../../src/common.dart';
-import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/linux/linux_doctor_test.dart:12]: Getting ready to migrate `fake_process_manager`.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**